src/generic/GenericTypes.jl

```diff
@@ -798,6 +798,46 @@ end
 
 const PuiseuxSeriesElem{T} = Union{PuiseuxSeriesRingElem{T}, PuiseuxSeriesFieldElem{T}} where T <: RingElement
 
+###############################################################################
+#
+#   AbsMSeriesRing / AbsMSeries
+#
+###############################################################################
+
+mutable struct AbsMSeriesRing{T <: RingElement, S} <:
+                                                 AbstractAlgebra.MSeriesRing{T}
+   base_ring::Ring
```

Comment:
Can't this be deleted as redundant, being equal to `base_ring(poly_ring)` (`poly_ring` the next field of the struct) ?

Reply:
I guess so.